GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -64,7 +76,7 @@ namespace gdjs {
 
     constructor(howl: Howl, volume: float, loop: boolean, rate: float) {
       this._howl = howl;
-      this._volume = volume;
+      this._volume = clampVolume(volume);
```

Comment:
nice. This seems indeed safer to clamp to the volume the same way it is done for the rate.

I'm not sure what's cleanest though.
- storing the clamped value (like you did for the volume)
- storing the original value, but clamp it when playing in the .play() method (like I did for the rate, L104)

In any case, we should do the same for both

Reply:
That's a good remark. I've tested on the current version, with a sound starting at volume 0 and an action increasing the volume at each frame, and another action displaying the volume inside a text:

The volume gets "clamped" by Howler, in the sense that it reaches 0.999... and then when you try to increase it a bit more (to 1.0000001 for example), Howler rejects it. So the maximum value stays there.
It would act similarly in the other direction (you would get clamped at 0.01...).

In both cases:
- it means the current GDevelop version "properly" clamp the volume between 0 and 1 for sounds (i.e: between 0 and 100 for events). At least it's not allowing this value to go outside the bounds! So it means we must store the clamped value (to avoid allowing a volume of 200, 300, 400... or -0.5, -1, -100...)

Then for the question of clamping the rate:
- it's currently not clamped (you can have a pitch/rate of 10 and it's a valid value). But for technical reasons, Howler can't go higher than 4 it seems - so we clamp it but we store the original value (because in the future another audio library could maybe play sounds faster).

So I think we are good in terms of behavior?

Reply:
Really good testing 👍 
Indeed, I think we're good in terms of behavior.
I would maybe add a comment in the code to explain why this architecture and avoid someone to refactor it in the future.
Either on the property, the setter or the constructor (or all of them 🙃)

Reply:
Great idea, I'll add comments so this is an explicit behavior.